include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -192,40 +178,82 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 
-						auto localOutPos = newReadBlockPos * blockDims;
 						auto* inDataAdress = inData + readBlockArrayOffset;
-						const size_t scratchOffset = asset::IImage::SBufferCopy::getLocalByteOffset(core::vector3du32_SIMD(localOutPos.x, localOutPos.y, localOutPos.z), scratchByteStrides);
-						decodeType* outDataAdress = (decodeType*)(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
-
-						constexpr uint8_t maxPlanes = 4;
 						const void* inSourcePixels[maxPlanes] = { inDataAdress, nullptr, nullptr, nullptr };
 
-						for (auto blockY = 0u; blockY < blockDims.y; blockY++)
-							for (auto blockX = 0u; blockX < blockDims.x; blockX++)
-							{
-								decodeType decodeBuffer[maxChannels] = {};
+						if constexpr (ExclusiveMode)
+						{
+							auto movedLocalOutPos = localOutPos + movingExclusiveVector;
+							const auto isSatMemorySafe = (movedLocalOutPos < core::vectorSIMDu32(state->extent.width, state->extent.height, state->extent.depth, movedLocalOutPos.w + 1)); // force true on .w
+							const auto doesItMoveOnYZorXZorXY = (localOutPos < movingOnYZorXZorXYCheckingVector);
 
-								asset::decodePixelsRuntime(inFormat, inSourcePixels, decodeBuffer, blockX, blockY);
-								memcpy(outDataAdress, decodeBuffer, scratchTexelByteSize);
+							if (isSatMemorySafe.all())
+							{
+								for (auto blockY = 0u; blockY < blockDims.y; blockY++)
+									for (auto blockX = 0u; blockX < blockDims.x; blockX++)
+									{
+										decodeType decodeBuffer[maxChannels] = {};
+
+										if (doesItMoveOnYZorXZorXY.any())
+										{
+											const size_t offset = asset::IImage::SBufferCopy::getLocalByteOffset(core::vector3du32_SIMD(localOutPos.x + blockX, localOutPos.y + blockY, localOutPos.z), scratchByteStrides);
+											memcpy(reinterpret_cast<uint8_t*>(scratchMemory) + offset, decodeBuffer, scratchTexelByteSize);
+										}
+										else
+										{
+											asset::decodePixelsRuntime(inFormat, inSourcePixels, decodeBuffer, blockX, blockY);
+											const size_t movedOffset = asset::IImage::SBufferCopy::getLocalByteOffset(core::vector3du32_SIMD(movedLocalOutPos.x + blockX, movedLocalOutPos.y + blockY, movedLocalOutPos.z), scratchByteStrides);
+											memcpy(reinterpret_cast<uint8_t*>(scratchMemory) + movedOffset, decodeBuffer, scratchTexelByteSize);
+										}
+									}
 							}
+						}
+						else
+						{
+							for (auto blockY = 0u; blockY < blockDims.y; blockY++)
+								for (auto blockX = 0u; blockX < blockDims.x; blockX++)
+								{
+									decodeType decodeBuffer[maxChannels] = {};
+
+									asset::decodePixelsRuntime(inFormat, inSourcePixels, decodeBuffer, blockX, blockY);
+									const size_t offset = asset::IImage::SBufferCopy::getLocalByteOffset(core::vector3du32_SIMD(localOutPos.x + blockX, localOutPos.y + blockY, localOutPos.z), scratchByteStrides);
+									memcpy(reinterpret_cast<uint8_t*>(scratchMemory) + offset, decodeBuffer, scratchTexelByteSize);
```

Comment:
decode straight to `reinterpret_cast<uint8_t*>(scratchMemory) + offset` dont use that silly `decodeBuffer`